omni-relayer/src/utils/near.rs

```diff
@@ -0,0 +1,117 @@
+        .iter()
+        .flat_map(|outcome| outcome.execution_outcome.outcome.logs.clone())
+        .filter_map(|log| serde_json::from_str::<Nep141LockerEvent>(&log).ok())
+        .collect::<Vec<_>>();
+
+    for log in nep_locker_event_logs {
+        info!("Processing Nep141LockerEvent: {:?}", log);
+
+        match log {
+            Nep141LockerEvent::InitTransferEvent {
+                ref transfer_message,
+            } => {
+                // TODO: If fee is insufficient, it should be handled later. For example,
+                // add to redis and try again in 1 hour
+                match utils::price::is_fee_sufficient(
+                    jsonrpc_client,
+                    &transfer_message.sender,
+                    &transfer_message.recipient,
+                    &transfer_message.token,
+                    transfer_message.fee.into(),
+                )
+                .await
+                {
+                    Ok(res) => {
+                        if !res {
+                            warn!("Fee is insufficient");
+                        }
+                    }
+                    Err(err) => {
+                        warn!("Failed to check fee: {}", err);
+                    }
+                }
+
+                utils::redis::add_event(
+                    redis_connection,
+                    &config.redis.near_init_transfer_events,
+                    transfer_message.origin_nonce.0.to_string(),
+                    log,
+                )
+                .await;
+            }
+            Nep141LockerEvent::SignTransferEvent {
+                ref message_payload,
+                ..
+            } => {
+                utils::redis::add_event(
+                    redis_connection,
+                    &config.redis.near_sign_transfer_events,
+                    message_payload.nonce.0.to_string(),
+                    log,
+                )
+                .await;
+            }
+            Nep141LockerEvent::FinTransferEvent { .. }
+            | Nep141LockerEvent::UpdateFeeEvent { .. } => todo!(),
+        }
+    }
+}
+
+fn find_nep_locker_event_outcomes(
+    config: &config::Config,
+    streamer_message: &StreamerMessage,
+) -> Vec<IndexerExecutionOutcomeWithReceipt> {
+    streamer_message
+        .shards
+        .iter()
+        .flat_map(|shard| shard.receipt_execution_outcomes.iter())
+        .filter(|outcome| is_nep_locker_event(config, &outcome.receipt))
+        .cloned()
+        .collect()
+}
+
+fn is_nep_locker_event(config: &config::Config, receipt: &ReceiptView) -> bool {
+    receipt.receiver_id == config.testnet.token_locker_id
+        && matches!(
+            receipt.receipt,
+            ReceiptEnumView::Action { ref actions, .. } if actions.iter().any(|action| {
+                matches!(action, ActionView::FunctionCall { method_name, .. } if method_name == "ft_on_transfer" || method_name == "sign_transfer_callback")
```

Comment:
"update_transfer_fee", "fin_transfer_callback". 

Should we event filter by method_name?

Reply:
> "update_transfer_fee", "fin_transfer_callback"

But relayer has nothing to do with these calls yet. `update_transfer_fee` doesn't require any action from relayer, as far as I know

Reply:
require. You should check fee one more time and handle transaction. In second case it can be transfer from Eth->Solana for example.